### PR TITLE
ci: optimize diff workflow

### PR DESCRIPTION
**Pull request by LingyuCoder**
## Summary

1. Use bash to upload diff report to rspack-report-website
2. Remove reports older than one month
3. Fix download binding bugs

## Require Documentation?

- [x] No
- [ ] Yes, the corresponding rspack-website PR is \_\_

### CODE REVIEW

**Comment by LingyuCoder** [comment]
!diff

File: .github/workflows/diff.yml [hunk 1]
```diff
@@ -3,6 +3,8 @@
 name: Diff Assets
```
Comment (LingyuCoder) [inline .github/workflows/diff.yml:3]: !diff

File: .github/workflows/diff.yml [hunk 2]
```diff
@@ -3,6 +3,8 @@
...
```
Comment (LingyuCoder) [inline .github/workflows/diff.yml:4]: !diff

File: .github/workflows/diff.yml [hunk 3]
```diff
@@ -3,6 +3,8 @@
...
 
 on:
+  pull_request_review_comment:
```
Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

File: .github/workflows/diff.yml [hunk 4]
```diff
@@ -14,67 +16,106 @@ jobs:
...
           API_TOKEN_GITHUB: ${{ secrets.RSPACK_REPORT_ACCESS_TOKEN }}
-        with:
-          source_file: diff_output/.
-          destination_repo: web-infra-dev/rspack-report-website
-          destination_folder: diff/${{ github.run_id }}/
-          user_email: lingyucoder@gmail.com
-          user_name: LingyuCoder
-          commit_message: Auto upload diff report
+        run: |
+          temp_dir=$(mktemp -d)
```
Comment (jerrykingxyz) [inline .github/workflows/diff.yml:66]: mktemp always return a new empty directory, maybe we can set the cache_dir="$HOME/.cache/diff_upload"

File: .github/workflows/diff.yml (hunk 3 above)
Comment (LingyuCoder) [inline .github/workflows/diff.yml:6]: !diff

**Pull request merged**